Cargo.toml

```diff
@@ -68,16 +72,18 @@ assert_cmd = "2.0.8"
 form_urlencoded = "1.0.1"
 indoc = "2.0"
 predicates = "3.0"
-hyper = { version = "0.14", features = ["server"] }
+hyper = { version = "1.2", features = ["server"] }
 tokio = { version = "1", features = ["rt", "sync", "time"] }
 tempfile = "3.2.0"
+hyper-util = { version = "0.1.3", features = ["server"] }
+http-body-util = "0.1.1"
 
 [features]
-# Ideally network-interface would be disabled by default on certain platforms
-# However: https://github.com/rust-lang/cargo/issues/1197
-default = ["online-tests", "rustls", "network-interface"]
+default = ["online-tests", "rustls"]
 native-tls = ["reqwest/native-tls", "reqwest/native-tls-alpn"]
 rustls = ["reqwest/rustls-tls", "reqwest/rustls-tls-webpki-roots", "reqwest/rustls-tls-native-roots"]
+
+# To be used by platforms that don't support binding to interface via SO_BINDTODEVICE
 network-interface = ["dep:network-interface"]
```

Comment:
Can we limit this so the dependency isn't pulled in on Linux (etc) even with the feature enabled?

Reply:
`network-interface` now shows up in `cargo tree --target x86_64-pc-windows-gnu` but not `cargo tree` so I think this works.